pass through LF compiler option

for the tsc option:

```
--newLine NEWLINE                  Specifies the end of line sequence to be used when emitting files: 'CRLF' (dos) or 'LF' (unix).
```

I added this to my tsconfig.json:

```
"compilerOptions": {
    "newLine": "LF"
```

however i still see lots of this on each commit of generated files:

```
warning: CRLF will be replaced by LF in bot1/src/mongo/MongoWrap-spec.js.
The file will have its original line endings in your working directory.
```

do these compiler options have to be explicitly passed through to tsc?

I added `"foo": "bar"` within atom, and got no errors, but running `tsc` at the command line did give:

    error TS5023: Unknown compiler option 'foo'.

so I assume they are NOT being passed from atom and have to be added to atom-typescript somewhere...?

--- a/lib/main/lang/core/languageServiceHost2.ts
+++ b/lib/main/lang/core/languageServiceHost2.ts
@@ -1,6 +1,7 @@
 import path = require('path');
 import utils = require('../utils');
 import fs = require('fs');
+import os = require('os')
 import textBuffer = require('basarat-text-buffer');
 
 import tsconfig = require('../../tsconfig/tsconfig');
@@ -328,6 +329,18 @@ export class LanguageServiceHost implements ts.LanguageServiceHost {
     ////////////////////////////////////////
 
     getCompilationSettings = () => this.config.project.compilerOptions;
+    getNewLine = () => {
+        let eol = os.EOL;
+        switch (this.config.project.compilerOptions.newLine) {
+            case ts.NewLineKind.CarriageReturnLineFeed:
+                eol = "\r\n";
+                break;
+            case ts.NewLineKind.LineFeed:
+                eol = "\n";
+                break;
+        }
+        return eol;
+    }
     getScriptFileNames = (): string[]=> Object.keys(this.fileNameToScript);
     getScriptVersion = (fileName: string): string => {
         var script = this.fileNameToScript[fileName];

--- a/lib/main/tsconfig/tsconfig.ts
+++ b/lib/main/tsconfig/tsconfig.ts
@@ -297,7 +297,9 @@ function rawToTsCompilerOptions(jsonOptions: CompilerOptions, projectDir: string
     var compilerOptions = <ts.CompilerOptions>mixin({}, defaults);
     for (var key in jsonOptions) {
         if (typescriptEnumMap[key]) {
-            compilerOptions[key] = typescriptEnumMap[key][jsonOptions[key].toLowerCase()];
+            let name = jsonOptions[key];
+            let map = typescriptEnumMap[key];
+            compilerOptions[key] = map[name.toLowerCase()] || map[name.toUpperCase()];
         }
         else {
             compilerOptions[key] = jsonOptions[key];

